src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java

```diff
@@ -557,28 +561,47 @@ private void checkClosed() throws SQLServerException {
-                        ResultSet.CONCUR_READ_ONLY);
                 String sProc = parseProcIdentifier(st.procedureName);
-                if (con.isKatmaiOrLater())
-                    rsProcedureMeta = s.executeQueryInternal("exec sp_sproc_columns_100 " + sProc + ", @ODBCVer=3");
-                else
-                    rsProcedureMeta = s.executeQueryInternal("exec sp_sproc_columns " + sProc + ", @ODBCVer=3");
-
-                // if rsProcedureMeta has next row, it means the stored procedure is found
-                if (rsProcedureMeta.next()) {
-                    procedureIsFound = true;
-                } else {
-                    procedureIsFound = false;
-                }
-                rsProcedureMeta.beforeFirst();
-
-                // Sixth is DATA_TYPE
-                rsProcedureMeta.getColumn(6).setFilter(new DataTypeFilter());
-                if (con.isKatmaiOrLater()) {
-                    rsProcedureMeta.getColumn(8).setFilter(new ZeroFixupFilter());
-                    rsProcedureMeta.getColumn(9).setFilter(new ZeroFixupFilter());
-                    rsProcedureMeta.getColumn(17).setFilter(new ZeroFixupFilter());
+                // DO NOT close the Statement as resultSet 'rsProcedureMeta' is required by other APIs
+                try (SQLServerStatement s = (SQLServerStatement) con.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE,
+                        ResultSet.CONCUR_READ_ONLY);
+                        SQLServerResultSet rsProcedureMeta = s.executeQueryInternal(
+                                con.isKatmaiOrLater() ? "exec sp_sproc_columns_100 " + sProc + ", @ODBCVer=3"
+                                                      : "exec sp_sproc_columns " + sProc + ", @ODBCVer=3")) {
+
+                    // if rsProcedureMeta has next row, it means the stored procedure is found
+                    if (rsProcedureMeta.next()) {
+                        procedureIsFound = true;
+                    } else {
+                        procedureIsFound = false;
+                    }
+
+                    rsProcedureMeta.beforeFirst();
+
+                    // Sixth is DATA_TYPE
+                    rsProcedureMeta.getColumn(6).setFilter(new DataTypeFilter());
+                    if (con.isKatmaiOrLater()) {
+                        rsProcedureMeta.getColumn(8).setFilter(new ZeroFixupFilter());
+                        rsProcedureMeta.getColumn(9).setFilter(new ZeroFixupFilter());
+                        rsProcedureMeta.getColumn(17).setFilter(new ZeroFixupFilter());
+                    }
+
+                    procMetadata = new ArrayList<>();
+
+                    // Process ResultSet Procedure Metadata for API usage
+                    while (rsProcedureMeta.next()) {
+                        procMetadata.add(new HashMap<String, Object>() {
+                            {
+                                put("DATA_TYPE", rsProcedureMeta.getShort("DATA_TYPE"));
```

Comment:
I think we should have some kind of an exception handling here. 
If one of the `ResultSet` getters fails here, the exception message will be misleading. The `catch` block you removed below can be moved up here.

Reply:
There is a catch block in this method at the end. The exception handling that was done for these getters is done same way here.

Reply:
Not really, the catch block will just re-throw the exception. 
```
catch (SQLServerException e) {
    throw e;
}
```

Reply:
So what's the problem with that? We don't need to wrap SQLServerException again!